### DISCUSSION TITLE
control HFS via tray icon

### DISCUSSION THREAD

**Discussion by rejetto** [rejetto/hfs#830] [Ideas]
this is meant to be a continuation of the sub-discussion at https://github.com/rejetto/hfs/discussions/282#discussioncomment-11387874

**Comment by rejetto** [comment]
@f2065
I just made a first version of the plugin, that is actually hiding the console by making a request to http://localhost/~/api/_hideConsole
and of course i also made _showConsole .

The idea is to include a separate exe with its tray icon. The plugin will launch such exe, like "other.exe -u=http://localhost" so that it knows where to make http requests.
Does it sound good?
I will share my work soon.

**Comment by f2065** [comment]
I solved this problem.
I launch hfs.exe from my process, specify the necessary console, and then track child processes and their windows.
Now I'm thinking how to track the ulpoad log - intercept the existing access.log or embed a plugin.
And how to find out (for the initial auto-configuration) where hfs.exe is (now I look in the registry HKEY_CURRENT_USER\Software\Classes\*\shell\AddToHFS3, but it may not be there).
The release will be soon:
![hfs3tt_a](https://github.com/user-attachments/assets/d05c76af-7908-4fe2-bca0-9eaa53c165ba)
![hfs3tt_b](https://github.com/user-attachments/assets/0ad45d5d-4c5b-407e-bdd4-335af34ac8c8)

**Comment by rejetto** [comment]
your project looks nice!
since you don't need anymore my part to hide the console, i [released my work as a plugin](https://github.com/rejetto/hfs/discussions/831).
anyway, i'm not sure if you'll still be able to track the child process after an update. The update process starts another process, and then another (2 restarts).

for the upload tracking, you can do the same as the admin-panel/logs. Try launching this at command line `curl localhost/~/api/get_log`

**Comment by rejetto** [comment]
@f2065 do you plan to release your project?

**Reply by f2065** [reply]
Yes, it will be published in the few days.
Is the config always in C:\Users\(profile)\.hfs\config.yaml ?
Or can it be in other places?
I need to know the HTTP and HTTPS port numbers, they may not be standard, I read them from the config.

**Reply by rejetto** [reply]
it's in the README https://github.com/rejetto/hfs?tab=readme-ov-file#where-is-it-stored
